instance-scheduler/main.go

```diff
@@ -38,38 +38,47 @@ aws ssm get-parameter --name environment_management_arn --with-decryption --prof
 aws secretsmanager get-secret-value --secret-id environment_management --profile mod --region eu-west-2
 */
 
-func getParameter(cfg aws.Config, parameterName string) string {
-	client := ssm.NewFromConfig(cfg)
-	input := &ssm.GetParameterInput{
+/*
+Interface that defines the set of Amazon SSM API operations required by the getParameter function.
+ISSMGetParameter is satisfied byt the Amazon SSM client's GetParameter method.
+*/
+type ISSMGetParameter interface {
+	GetParameter(ctx context.Context, params *ssm.GetParameterInput, optFns ...func(*ssm.Options)) (*ssm.GetParameterOutput, error)
+}
+
+func getParameter(client ISSMGetParameter, parameterName string) string {
+	result, err := client.GetParameter(context.TODO(), &ssm.GetParameterInput{
 		Name:           aws.String(parameterName),
 		WithDecryption: aws.Bool(true),
-	}
-	result, err := client.GetParameter(context.TODO(), input)
+	})
 	if err != nil {
 		log.Fatal(err)
 	}
 	return *result.Parameter.Value
 }
 
-func getSecret(cfg aws.Config, secretId string) string {
-	client := secretsmanager.NewFromConfig(cfg)
-	input := &secretsmanager.GetSecretValueInput{
+/*
+Interface that defines the set of Amazon secretsmanager API operations required by the getSecret function.
+ISecretManagerGetSecretValue is satisfied byt the Amazon secretsmanager client's GetSecretValue method.
+*/
+type ISecretManagerGetSecretValue interface {
+	GetSecretValue(ctx context.Context, params *secretsmanager.GetSecretValueInput, optFns ...func(*secretsmanager.Options)) (*secretsmanager.GetSecretValueOutput, error)
+}
+
+func getSecret(client ISecretManagerGetSecretValue, secretId string) string {
+	result, err := client.GetSecretValue(context.TODO(), &secretsmanager.GetSecretValueInput{
 		SecretId:     aws.String(secretId),
 		VersionStage: aws.String("AWSCURRENT"),
-	}
-	result, err := client.GetSecretValue(context.TODO(), input)
+	})
+
 	if err != nil {
 		log.Fatal(err)
 	}
 	return *result.SecretString
 }
 
-func getNonProductionAccounts(cfg aws.Config, skipAccountNames string) map[string]string {
+func getNonProductionAccounts(secretId string, environments string, skipAccountNames string) map[string]string {
```

Comment:
It doesn't look like secretId is being used in this function. Is it for future-proofing?

Reply:
good catch, going to fix it